templates/buildkite-elastic.yml

```diff
@@ -185,9 +185,9 @@ Resources:
         config:
           commands:
             01-download-github-key:
-              test: test -n "$(ProvisionBucket)"
+              test: test -n "$(SecretsBucket)"
               command: |
-                aws s3 cp s3://$(ProvisionBucket)/id_rsa_buildkite /root/.ssh/id_rsa
+                aws s3 cp s3://$(SecretsBucket)/id_rsa_buildkite /root/.ssh/id_rsa
```

Comment:
Is the plan to updated or removed this? The `aws s3 cp …` command without `--sse-c` will fail for a client-encrypted object;

```
$ aws s3 cp s3://pda-test/payload ./
A client error (400) occurred when calling the HeadObject operation: Bad Request

$ aws s3 cp --sse-c --sse-c-key "$key" s3://pda-test/payload ./
download: s3://pda-test/payload to ./payload
```

And if it did successfully download, the secret will be unencrypted at rest.

Is the plan to download the secrets from S3 for each job? That seems to be the only way to use S3 client-side encryption here. Or should the encryption/decryption be done completely external to S3? Or does “encrypted at rest” refer to “at rest on S3” and not on the agent's local disk?